src/openvic-simulation/pop/Pop.cpp

```diff
@@ -162,6 +172,59 @@ void Pop::update_gamestate(
 //TODO store income
 void Pop::add_rgo_owner_income(const fixed_point_t income) {}
 void Pop::add_rgo_worker_income(const fixed_point_t income) {}
+//adds income to the temporary stores and then to general income store
+void Pop::add_artisanal_income(const fixed_point_t income){
+	artisanal_income += income;
+	this->income += income;
+}
+void Pop::add_factory_worker_income(const fixed_point_t income){
+	this->factory_worker_income += income;
+	this->income += income;
+}
+void Pop::add_factory_owner_income(const fixed_point_t income){
+	this->factory_owner_income += income;
+	this->income += income;
+}
+void Pop::add_unemployment_subsidies_income(const fixed_point_t income){
+	this->unemployment_subsidies_income += income;
+	this->income += income;
+}
+void Pop::add_pensions_income(const fixed_point_t income){
+	this->pensions_income += income;
+	this->income += income;
+}
+void Pop::add_government_salary_administration_income(const fixed_point_t income){
+	this->government_salary_administration_income += income;
+	this->income += income;
+}
+void Pop::add_government_salary_education_income(const fixed_point_t income){
+	this->government_salary_education_income += income;
+	this->income += income;
+}
+void Pop::add_government_salary_military_income(const fixed_point_t income){
+	this->government_salary_military_income += income;
+	this->income += income;
+}
+void Pop::add_event_and_decision_income(const fixed_point_t income){
+	this->event_and_decision_income += income;
+	this->income += income;
+}
+void Pop::add_loan_interest_payments(const fixed_point_t income){
+	this->loan_interest_payments += income;
+	this->income += income;
+}
+void Pop::clear_income_per_source(){ //resets all of the seprate stores
+	this->artisanal_income = fixed_point_t(0);
+	this->factory_worker_income = fixed_point_t(0);
+	this->factory_owner_income = fixed_point_t(0);
```

Comment:
Replace all uses of `fixed_point_t(0)` with `fixed_point_t::_0()`, using a fixed constant rather than going through the normal constructor